pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -323,68 +322,35 @@ func createTargetConfigController(ctx context.Context, syncCtx factory.SyncConte
 	return false, nil
 }
 
-// setCloudControllerOwnerCondition sets the condition to False if either external cloud
-// provider has been successfully applied for all static pods or it's not set at all. Otherwise
-// it sets the condition to True.
-func setCloudControllerOwnerCondition(ctx context.Context, c TargetConfigController) error {
+// clearCloudControllerOwnerCondition removes the CloudControllerOwner condition if it exists.
+// Prior to version 4.15 of OpenShift, this condition was used to signal the ownership of the
+// cloud controllers. After 4.15 this condition is no longer needed as the external cloud
+// controllers are now the default.
+func removeCloudControllerOwnerCondition(ctx context.Context, c TargetConfigController) error {
```

Comment:
Can you please add a TODO here that this should be removed in 4.17?

Also, there seems to be other cloud provider bits that should be removed? Is this planned?

Reply:
oh I see this is being worked on in https://github.com/openshift/cluster-kube-controller-manager-operator/pull/799

Reply:
@atiratree did you still want me to ad a TODO here?

Reply:
yes please